src/test/common/test_counter.cc

```diff
@@ -0,0 +1,39 @@
+#include "common/DecayCounter.h"
+
+#include <gtest/gtest.h>
+
+#include <list>
+#include <cmath>
+
+TEST(DecayCounter, steady)
+{
+  static const double duration = 2.0;
+  static const double max = 2048.0;
+  static const double rate = 3.5;
+
+  DecayCounter d{DecayRate{rate}};
+  d.hit(max);
+  const auto start = DecayCounter::clock::now();
+  double total = 0.0;
+  while (1) {
+    const auto now = DecayCounter::clock::now();
+    auto el = std::chrono::duration<double>(now-start);
+    if (el.count() > duration) {
+      break;
+    }
+
+    double v = d.get();
+    double diff = max-v;
+    if (diff > 0.0) {
+      d.hit(diff);
+      total += diff;
+    }
+  }
+
+  /* Decay function: dN/dt = -λN where λ = ln(0.5)/rate
+   * Integrating over t: N = -λN*t (+c)
```

Comment:
Shouldn't the solution of this ODE be exponential? i.e N = N_init* e^(- λ*t)

Reply:
If N changes with time, yes. That's the solution to the ODE. If we fix N (because we keep "refiling" the counter to N_0), then it does not change with time. The variable names are confusing in that regard so I'll correct that.

Reply:
Ahh got it. Thanks!